src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -1216,18 +1216,19 @@ static void window_tile_inspector_dropdown(rct_window* w, rct_widgetindex widget
     }
 }
 
-static void window_tile_inspector_tool_update(rct_window* w, rct_widgetindex widgetIndex, int32_t x, int32_t y)
+static void window_tile_inspector_tool_update(rct_window* w, rct_widgetindex widgetIndex, ScreenCoordsXY screenCoords)
 {
     map_invalidate_selection_rect();
 
     gMapSelectFlags |= MAP_SELECT_FLAG_ENABLE;
 
-    int16_t mapX = x;
-    int16_t mapY = y;
+    int16_t mapX = screenCoords.x;
```

Comment:
This is horrible. We should refactor this.

mapX and mapY are CoordsXY values but you need to set them to screenCoords.x and screenCoords.y if clickedElement is nullptr below as `screen_pos_to_map_pos` is a horrible function that takes in screen coords and spits out coordsxy to the same variable.

Reply:
I'm assuming this is part of the refactoring and nothing needs to be done here, for now?

Reply:
Yeah I didn't realise when I wrote this how prevalent the abuse caused by `screen_pos_to_map_pos` was. Leave for now then fix in next refactoring.